components/eamxx/src/physics/ml_correction/eamxx_ml_correction_process_interface.cpp

```diff
@@ -22,40 +28,90 @@ void MLCorrection::set_grids(
   const auto &grid_name = m_grid->name();
   m_num_cols = m_grid->get_num_local_dofs();  // Number of columns on this rank
   m_num_levs =
       m_grid->get_num_vertical_levels();  // Number of levels per column
-
   // Define the different field layouts that will be used for this process
 
   // Layout for 3D (2d horiz X 1d vertical) variable defined at mid-level and
   // interfaces
   FieldLayout scalar3d_layout_mid{{COL, LEV}, {m_num_cols, m_num_levs}};
 
-  // Set of fields used strictly as input
-  add_field<Required>("qv", scalar3d_layout_mid, Q, grid_name, "tracers");
-
-  // Set of fields used strictly as output
-  add_field<Computed>("qv_nudging_tend", scalar3d_layout_mid, Q, grid_name);
-
-  // Set of fields used as input and output
-  // - There are no fields used as both input and output.
+  /* ----------------------- WARNING --------------------------------*/
+  /* The following is a HACK to get things moving, we don't want to
+   * add all fields as "updated" long-term.  A separate stream of work
+   * is adapting the infrastructure to allow for a generic "add_field" call
+   * to be used here which we can then setup using the m_fields_ml_output_variables variable
+   */
+  add_field<Updated>("T_mid", scalar3d_layout_mid, K, grid_name, ps);
+  add_field<Updated>("qv",    scalar3d_layout_mid, Q, grid_name, "tracers", ps);
+  add_field<Updated>("u",     scalar3d_layout_mid, m/s, grid_name, ps);
+  add_field<Updated>("v",     scalar3d_layout_mid, m/s, grid_name, ps);
+  /* ----------------------- WARNING --------------------------------*/
+  add_group<Updated>("tracers", grid_name, 1, Bundling::Required);
 }
 
-// =========================================================================================
 void MLCorrection::initialize_impl(const RunType /* run_type */) {
-  // Nothing to do
+  // Do nothing
 }
 
 // =========================================================================================
-void MLCorrection::run_impl(const double /* dt */) {
-  auto qv         = get_field_in("qv").get_view<const Real **>();
-  auto qv_nudging = get_field_out("qv_nudging").get_view<Real **>();
+void MLCorrection::run_impl(const double dt) {
+  namespace py      = pybind11;
+  // use model time to infer solar zenith angle for the ML prediction
+  auto current_ts = timestamp();
+  std::string datetime_str = current_ts.get_date_string() + " " + current_ts.get_time_string();
+  const auto &qv_field = get_field_out("qv");
+  const auto &qv       = qv_field.get_view<Real **, Host>();
+  const auto &T_mid_field = get_field_out("T_mid");
+  const auto &T_mid       = T_mid_field.get_view<Real **, Host>();
+  const auto &u_field = get_field_out("u");
+  const auto &u       = u_field.get_view<Real **, Host>();
+  const auto &v_field = get_field_out("v");
+  const auto &v       = v_field.get_view<Real **, Host>();
+
+  // having m_lat and m_lon in set_grids breaks the standalone unit test
+  m_lat  = m_grid->get_geometry_data("lat");
```

Comment:
The geometry data fields should really be grabbed just once during `set_grids`.

Reply:
Is there an example of this working alongside the unit test using mesh free grid? I originally had it in `set_grids` but it was breaking the unit test so I had to move it.

Reply:
Odd. The (standalone) unit tests do have some peculiarity, since they don't have dynamics building the grids. But if the geo data was set at all in the grid, then it should already be available during set_grid. In other words, if it failed there, it should fail here too...

Reply:
This is the error when I move these lines to `set_grids`:
```
  abstract_grid.cpp:173: FAIL:
  has_geometry_data(name)
  Error! Geometry data 'lat' not found.
```